lib/src/wiggle_abi/dictionary_impl.rs

```diff
@@ -9,20 +12,63 @@ use {
     wiggle::GuestPtr,
 };
 
+use memoize::memoize;
+
+#[memoize]
```

Comment:
I added `memoize` here to avoid reading the file multiple times.

I think a better solution may be to have the file get read into memory when the `FastlyDictionary.open` method is called instead of reading it when the `FastlyDictionary.get` method is called (which we currently do).

Reply:
> I added `memoize` here to avoid reading the file multiple times.

I reckon it would be nice if you could update the dictionary and see the changes without restarting Viceroy?

> I think a better solution may be to have the file get read into memory when the FastlyDictionary.open method is called instead of reading it when the FastlyDictionary.get method is called (which we currently do).

In real Compute@Edge, the dictionary is lazy-loaded on Get, more or less, so the existing behavior here seems most symmetric.

Reply:
Memoization was removed in [`5623140` (#61)](https://github.com/fastly/Viceroy/pull/61/commits/5623140308d3e62c9c8b7ae1778b05008efd34a7)